sky/utils/config_utils.py

```diff
@@ -199,6 +199,13 @@ def merge_k8s_configs(
                         else:
                             base_config[key].append(new_volume)
             else:
-                base_config[key].extend(value)
+                # For other list values, merge lists and maintain uniqueness.
+                # Preserve order by keeping original items and appending any
+                # new item not already present.
+                result = list(base_config[key])
+                for item in value:
+                    if item not in result:
```

Comment:
A list item might be treated as map in kubernetes (https://kubernetes.io/docs/reference/using-api/server-side-apply/#merge-strategy), in this case the equality check should be based on the `listMapKey`.